src/Symfony/Bridge/Doctrine/Messenger/DoctrineClearEntityManagerMiddleware.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Messenger;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+
+/**
+ * Clears entity manager after calling all handlers.
+ *
+ * @author Konstanton Myakshin <koc-dp@yandex.ru>
+ *
+ * @experimental in 4.4
+ */
+class DoctrineClearEntityManagerMiddleware implements MiddlewareInterface
+{
+    private $managerRegistry;
+    private $entityManagerName;
+
+    public function __construct(ManagerRegistry $managerRegistry, string $entityManagerName = null)
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->entityManagerName = $entityManagerName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+        $entityManager = $this->managerRegistry->getManager($this->entityManagerName);
+
+        if (!$entityManager instanceof EntityManagerInterface) {
+            throw new \InvalidArgumentException(sprintf('The ObjectManager with name "%s" must be an instance of EntityManagerInterface', $this->entityManagerName));
```

Comment:
This could throw https://github.com/symfony/symfony/blob/7596da1cdcc2eb06bd15c967b43e9351de0a39c2/src/Symfony/Component/Messenger/Exception/UnrecoverableMessageHandlingException.php#L24 as it does not make sense to retry in this case.

Reply:
done